plugin/pulpcore/plugin/download/asyncio/exceptions.py

```diff
@@ -0,0 +1,34 @@
+from gettext import gettext as _
+
+from pulpcore.exceptions import PulpException
+
+
+class DownloaderValidationError(PulpException):
+    """
+    A base class for all Download Validation Errors.
+    """
+    pass
+
+
+class DigestValidationError(DownloaderValidationError):
+    """
+    Raised when an Artifact fails to validate a digest checksum.
+    """
+
+    def __init__(self):
+        super().__init__("PLP0003")
+
+    def __str__(self):
+        return _("An Artifact failed validation due to checksum.")
+
+
+class SizeValidationError(DownloaderValidationError):
+    """
+    Raised when an Artifact fails to validate a size checksum.
+    """
+
+    def __init__(self):
+        super().__init__("PLP0004")
+
+    def __str__(self):
+        return _("An Artifact failed validation due to size.")
```

Comment:
What if the file being downloaded w/ validation is not an artifact?  Misleading?  Also, I think including the url would be helpful.

Reply:
same fix here as above.